packages/cozy-harvest-lib/src/services/budget-insight.js

```diff
@@ -505,6 +523,44 @@ const shouldResumeConnection = error => {
   )
 }
 
+/**
+ * Tries to find an existing account, associated to an existing trigger
+ * with the given bi connection id
+ *
+ * @param  {CozyClient} options.client - Cozy client
+ * @param  {Object} options.konnector - Konnector manifest
+ * @param  {Integer} options.connectionId - BI connection id
+ * @return {Account|undefined} An account with a trigger with the same identifier if any
+ */
+export const fetchAccountWithBiConnection = async ({
+  client,
+  konnector,
+  connectionId
+}) => {
+  const [accountsResult, triggersResult] = await Promise.all([
+    client.query(
+      Q('io.cozy.accounts')
+        .where({ data: { auth: { bi: { connId: connectionId } } } })
+        .indexFields(['data.auth.bi.connId'])
+    ),
+    client.query(
+      Q('io.cozy.triggers')
+        .where({
+          message: {
+            konnector: konnector.slug
+          }
+        })
+        .indexFields(['message.konnector'])
+    )
+  ])
+
+  const accountsIndex = keyBy(accountsResult.data, '_id')
+  const trigger = triggersResult.data.find(
+    t => accountsIndex[get(t, 'message.account')]
+  )
+  return trigger ? accountsIndex[trigger.message.account] : null
```

Comment:
nit: personnally I would have tested a case where
```
 else if (qdef.doctype === 'io.cozy.triggers') {
        return {
          data: [
            {
              message: { account: 'another_account_id' }
            }
          ]
        }
```
to be sure we go in the null case => not throwing